src/eva-gram/src/database/input_tool.py

```diff
@@ -0,0 +1,189 @@
+        add_user(cur, owner)
+    for experiment in sample_dataset['experiments']:
+        add_experiment(cur, experiment)
+    for plot in (sample_dataset['plots']):
+        add_plot(cur, plot, sample_dataset['observation_dirs'])
+
+
+def get_observation_name(obs_dirs, filename):
+    # TODO: resolve duplicate filenames from different observations
+    for observation in obs_dirs:
+        obs_dir_path = os.path.join(EXPERIMENT_DATA_PATH, observation)
+        for plot in os.listdir(obs_dir_path):
+            if filename == plot:
+                return observation
+    return None
+
+
+def insert_table_record(cur, data, table):
+    query = "INSERT INTO {} (".format(table)
+    query += ', '.join(data)
+    query += ") VALUES ("
+    query += ', '.join(["%s" for i in range(len(data))])
+    query += ")"
+
+    cur.execute(query, tuple(data.values()))
+
+
+def add_user(cur, user_obj):
+    # check not null constraints
+    # TODO: owner_id made optional for user to provide
+    if 'owner_id' not in user_obj or 'username' not in user_obj:
+        print("Skipping current user: needs both owner_id and username")
+        return 1
+    else:
+        insert_table_record(cur, user_obj, "owners")
+    return 0
+
+
+def delete_user(cur, username):
+    cur.execute("DELETE FROM owners WHERE username=%s", (username,))
+
+
+def add_experiment(cur, experiment_obj):
+    # check not null constraints
+    # TODO: experiment_id made optional for user to provide
+    if 'experiment_id' not in experiment_obj or 'owner_id' not in experiment_obj:
+        print("Skipping current experiment: must provide both experiment_id and owner_id")
+        return 1
+    else:
+        insert_table_record(cur, experiment_obj, "experiments")
+    return 0
+
+
+def add_plot(cur, plot_obj, observation_dirs):
+    # check not null constraints
+    # TODO: plot_id made optional for user to provide
+    if 'plot_id' not in plot_obj or 'plot_file' not in plot_obj or 'experiment_id' not in plot_obj or 'group_id' not in plot_obj or 'observation_id' not in plot_obj:
+        print(
+            "Skipping current plot: needs plot file, experiment, group, and observation ids")
+        return 1
+
+    plot_filename = plot_obj['plot_file']
+    observation_name = get_observation_name(observation_dirs, plot_filename)
+    plot_file_path = os.path.join(
+        EXPERIMENT_DATA_PATH, observation_name, plot_filename)
+
+    with open(plot_file_path, 'rb') as file:
+        dictionary = pickle.load(file)
+
+    # extract the div and script components
+    div = dictionary['div']
+    script = dictionary['script']
+
+    # parse filename for components variable name, channel, and group name
+    filename_no_extension = os.path.splitext(plot_filename)[0]
+    plot_components = filename_no_extension.split("_")
+
+    var_name = plot_components[0]
+    # parse channel fields for brightnessTemperature observations
+    if var_name == "brightnessTemperature":
```

Comment:
Might have been discussed before, but can file names reflect null channel instead of hardcoding the case into Python?  For example:

`specificHumidity__hofx-vs-gsihofx.pkl`
and note the double delimiter.